create_geometry_3D.py

```diff
@@ -0,0 +1,258 @@
+    print(" - Mesh has " + str(len(node_tags)) + " nodes and " + str(num_elem) +
+          " elements")
+
+    num_nodes = len(node_tags)
+    coord = np.reshape(node_coords, (num_nodes, 3))
+
+    if elem_types[0] == 1:
+        node_shape = 2
+        num_1d_elements = len(elem_tags[0])
+        node_tag_1D = np.reshape(elem_node_tags[0], (num_1d_elements, node_shape))
+
+    if elem_types[1] == 2:
+        node_shape = 3  # node_shape amount of nodes in element
+        num_2d_elements = len(elem_tags[1])
+        node_tag_2D = np.reshape(elem_node_tags[1], (num_2d_elements, node_shape))
+
+    if dims == 3:
+        if elem_types[2] == 4:
+            node_shape = 4
+            num_3d_elements = len(elem_tags[2])
+            node_tag_3D = np.reshape(elem_node_tags[2], (num_3d_elements, node_shape))
+        return coord, node_tags, elem_types, elem_tags, node_tag_1D, node_tag_2D, node_tag_3D
+
+    if dims == 2:
+        return coord, node_tags, elem_types, elem_tags, node_tag_1D, node_tag_2D
+
+
+def generate_gmsh_mesh(point_coordinates, point_pairs, depth, mesh_size, dims, save_file, name_label):
+    """
+    generates mesh for geometries in gmsh
+    :param point_coordinates: geometry points coordinates
+    :param point_pairs: points paired for lines
+    :param depth: depth of 3D geometry
+    :param mesh_size: mesh size
+    :param dims: geometry dimension (2=2D or 3=3D)
+    :param save_file: if True saves mesh data to mdpa files
+    :param name_label: surface name label from user input
+    :return: saves mesh data and opens gmsh interface
+    """
+    gmsh.initialize()
+    gmsh.model.add(mesh_output_name)
+
+    volumes = []
+    if dims == 3:
+        make_geometry_3D(point_coordinates, point_pairs, mesh_size, depth, name_label)
+        gmsh.model.geo.synchronize()
+        gmsh.model.mesh.generate(dims)
+        # extract mesh data
+        node_coords, node_tags, elem_types, elem_tags, node_tag_1D, node_tag_2D, node_tag_3D = extract_mesh_data(dims)
+        nodes, lines, surfaces = get_data_for_kratos(node_coords, node_tags, elem_tags, node_tag_1D, node_tag_2D)
+        volumes = np.concatenate((elem_tags[2][:, None], np.array(node_tag_3D)), axis=1)
+
+    if dims == 2:
+        make_geometry_2D(point_coordinates, point_pairs, mesh_size, name_label)
+        gmsh.model.geo.synchronize()
+        gmsh.model.mesh.generate(dims)
+        # extract mesh data
+        node_coords, node_tags, elem_types, elem_tags, node_tag_1D, node_tag_2D = extract_mesh_data(dims)
+        nodes, lines, surfaces = get_data_for_kratos(node_coords, node_tags, elem_tags, node_tag_1D, node_tag_2D)
+
+    if save_file is True:
+        save_mesh_data(nodes, lines, surfaces, volumes)
+
+    # writes mesh file output in .msh format
+    file_extension = ".msh"
+    mesh_output_file = mesh_output_name + file_extension
+    gmsh.write(mesh_output_file)
+
+    # opens GMsh interface
+    if gmsh_interface is True:
```

Comment:
this is not an input parameter, therefore this function won't work if you call it from another file than this file